### PR TITLE
LPD-41065 Exporting and Importing Objects Widgets Across Different Environments

### PR DESCRIPTION
**Pull request by vngoliveira**
[LPD-36658](https://liferay.atlassian.net/browse/LPD-36658)

[LPD-36658]: https://liferay.atlassian.net/browse/LPD-36658?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by vngoliveira** [comment]
Hey team!
I am sending these first changes to be reviewed while I am working on the integration test and the upgrade test

File: modules/apps/object/object-rest-test/src/testFunctional/macros/ObjectDefinitionAPI.macro [hunk 1]
```diff
@@ -261,6 +261,15 @@ definition {
...
 	}
 
+	@summary = "Default summary"
+	macro _getObjectDefinitionClassNameSuffix(objectDefinitionId = null) {
+		var objectDefinitionClassName = ObjectDefinitionAPI._getObjectDefinitionClassNameById(objectDefinitionId = ${objectDefinitionId});
+
+		var objectDefinitionClassNameSuffix = StringUtil.extractLast(${objectDefinitionClassName}, "#");
+
+		return ${objectDefinitionClassNameSuffix};
+	}
```
Comment (carolmariaabb) [inline modules/apps/object/object-rest-test/src/testFunctional/macros/ObjectDefinitionAPI.macro:271]: @vngoliveira could you please change to be `_getObjectDefinitionPortletId` returning:

```
return "com_liferay_object_web_internal_object_definitions_portlet_ObjectDefinitionsPortlet_${objectDefinitionClassNameSuffix}";
```
?

File: modules/apps/site-initializer/site-initializer-welcome/src/main/resources/site-initializer/layout-utility-page-entries/cookie-policy/page-definition.json [hunk 2]
```diff
@@ -258,7 +258,7 @@
 															"widgetInstance": {
 																"widgetConfig": {
 																},
-																"widgetName": "com_liferay_object_web_internal_object_definitions_portlet_ObjectDefinitionsPortlet_[$OBJECT_DEFINITION_ID:NecessaryCookieEntry$]"
+																"widgetName": "com_liferay_object_web_internal_object_definitions_portlet_ObjectDefinitionsPortlet_[$OBJECT_DEFINITION_CLASS_NAME_SUFFIX:NecessaryCookieEntry$]"
 															}
```
Comment (carolmariaabb) [inline modules/apps/site-initializer/site-initializer-welcome/src/main/resources/site-initializer/layout-utility-page-entries/cookie-policy/page-definition.json:262]: @vngoliveira same logic here, change from `OBJECT_DEFINITION_CLASS_NAME_SUFFIX` to `OBJECT_DEFINITION_PORTLET_ID`

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/BundleSiteInitializer.java [hunk 3]
```diff
@@ -1177,6 +1178,13 @@ private void _addObjectDefinitions(
...
+			stringUtilReplaceValues.put(
+				"OBJECT_DEFINITION_PORTLET_ID:" +
+					serviceBuilderObjectDefinition.getShortName(),
+				ObjectPortletKeys.OBJECT_DEFINITIONS +
+					StringUtil.split(
+						serviceBuilderObjectDefinition.getClassName(),
+						StringPool.POUND)[1]);
 			stringUtilReplaceValues.put(
 				"OBJECT_DEFINITION_ID:" +
 					serviceBuilderObjectDefinition.getShortName(),
```
Comment (carolmariaabb) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/BundleSiteInitializer.java:1190]: @vngoliveira `OBJECT_DEFINITION_PORTLET_ID` should be after `OBJECT_DEFINITION_ID` to follow alphabetical order

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/BundleSiteInitializer.java [hunk 4]
```diff
@@ -1248,6 +1256,11 @@ private void _addObjectDefinitions(
 			stringUtilReplaceValues.put(
 				"OBJECT_DEFINITION_CLASS_NAME:" + objectDefinition.getName(),
 				objectDefinition.getClassName());
+			stringUtilReplaceValues.put(
+				"OBJECT_DEFINITION_PORTLET_ID:" + objectDefinition.getName(),
+				ObjectPortletKeys.OBJECT_DEFINITIONS +
+					StringUtil.split(
+						objectDefinition.getClassName(), StringPool.POUND)[1]);
 			stringUtilReplaceValues.put(
```
Comment (carolmariaabb) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/BundleSiteInitializer.java:1264]: @vngoliveira same here

File: modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/BundleSiteInitializer.java
Comment (carolmariaabb) [inline modules/apps/site-initializer/site-initializer-extender/site-initializer-extender/src/main/java/com/liferay/site/initializer/extender/internal/BundleSiteInitializer.java:1]: @vngoliveira please create a private method to reuse code:

```
private void _addObjectDefinition(
	String objectDefinitionClassName, long objectDefinitionId,
	String objectDefinitionName,
	Map<String, String> stringUtilReplaceValues) {

	stringUtilReplaceValues.put(
		"OBJECT_DEFINITION_CLASS_NAME:" + objectDefinitionName,
		objectDefinitionClassName);
	stringUtilReplaceValues.put(
		"OBJECT_DEFINITION_ID:" + objectDefinitionName,
		String.valueOf(objectDefinitionId));
	stringUtilReplaceValues.put(
		"OBJECT_DEFINITION_PORTLET_ID:" + objectDefinitionName,
		ObjectPortletKeys.OBJECT_DEFINITIONS +
			StringUtil.split(objectDefinitionClassName, StringPool.POUND)
				[1]);
}
```

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

**Comment by vngoliveira** [comment]
ci:test:sf

**Comment by carolmariaabb** [comment]
@vngoliveira see https://github.com/liferay-bpm/liferay-portal/pull/3841

**Pull request closed without merging**